src/main/kotlin/org/jitsi/nlj/transform/node/incoming/Vp8Parser.kt

```diff
@@ -16,68 +16,84 @@
-        val videoRtpPacket: VideoRtpPacket = packetInfo.packet as VideoRtpPacket
-        if (videoRtpPacket is Vp8Packet) {
-            // If this was part of a keyframe, it will have already had it set
-            if (videoRtpPacket.height > -1) {
-                // TODO: handle case where new height is from a packet older than the
-                // latest height we've seen.
-                ssrcToHeight.putIfAbsent(videoRtpPacket.ssrc, videoRtpPacket.height)
-            } else {
-                videoRtpPacket.height = ssrcToHeight[videoRtpPacket.ssrc] ?: -1
-            }
-            if (videoRtpPacket.isKeyframe) {
-                logger.cdebug { "Received a keyframe for ssrc ${videoRtpPacket.ssrc} ${videoRtpPacket.sequenceNumber}" }
-                numKeyframes++
+    override fun observe(packetInfo: PacketInfo) {
+        val vp8Packet = packetInfo.packet as Vp8Packet
+        if (vp8Packet.height > -1) {
+            // TODO: handle case where new height is from a packet older than the
+            // latest height we've seen.
+            val enc = findRtpEncodingDesc(vp8Packet)
+            if (enc != null) {
+                val newLayers = enc.layers.map { layer -> RtpLayerDesc(layer, height = vp8Packet.height) }
+                enc.layers = newLayers.toTypedArray()
             }
+        }
+        if (vp8Packet.isKeyframe) {
+            logger.cdebug { "Received a keyframe for ssrc ${vp8Packet.ssrc} ${vp8Packet.sequenceNumber}" }
+            numKeyframes++
+        }
 
-            pictureIdState.setState(videoRtpPacket.hasPictureId, videoRtpPacket) {
-                "Packet Data: ${videoRtpPacket.toHex(80)}"
-            }
-            extendedPictureIdState.setState(videoRtpPacket.hasExtendedPictureId, videoRtpPacket) {
-                "Packet Data: ${videoRtpPacket.toHex(80)}"
-            }
-            tidWithoutTl0PicIdxState.setState(
-                videoRtpPacket.hasTL0PICIDX || !videoRtpPacket.hasTemporalLayerIndex, videoRtpPacket
-            ) {
-                "Packet Data: ${videoRtpPacket.toHex(80)}"
+        pictureIdState.setState(vp8Packet.hasPictureId, vp8Packet) {
+            "Packet Data: ${vp8Packet.toHex(80)}"
+        }
+        extendedPictureIdState.setState(vp8Packet.hasExtendedPictureId, vp8Packet) {
+            "Packet Data: ${vp8Packet.toHex(80)}"
+        }
+        tidWithoutTl0PicIdxState.setState(
+            vp8Packet.hasTL0PICIDX || !vp8Packet.hasTemporalLayerIndex, vp8Packet
+        ) {
+            "Packet Data: ${vp8Packet.toHex(80)}"
+        }
+    }
+
+    override fun handleEvent(event: Event) {
+        when (event) {
+            is SetMediaSourcesEvent -> {
+                sources = event.mediaSourceDescs
             }
         }
+        super.handleEvent(event)
+    }
 
-        return packetInfo
+    private fun findRtpEncodingDesc(packet: VideoRtpPacket): RtpEncodingDesc? {
+        for (source in sources) {
```

Comment:
nit: I think this could be something like:
```sources.find { it.findRtpEncodingDesc(packet.ssrc) != null }```

Reply:
No, because we're returning an RtpEncodingDesc from one of the sources, rather than just the source in which the desc in question lives.